### PR TITLE
CONSOLE-3241: Use cluster proxy for managed cluster API server requests

### PR DESCRIPTION
**Pull request by TheRealJon**
Update server to remove original multicluster proxying implementation and instead proxy managed cluster requests to the MCE cluster proxy service

### CODE REVIEW

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

**Comment by TheRealJon** [comment]
/retest

Timed out

**Comment by TheRealJon** [comment]
/retest

Panic in openshift operator controller manager pod. Reported in Slack.

**Comment by TheRealJon** [comment]
/retest

Cypress verification failed

**Comment by yapei** [comment]
@TheRealJon wheh viewing manged cluster pages, it seems managed cluster pages can not be loaded successfully and I can see some errors in console log
```
I0920 10:01:32.237240 1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/uscluster/apis/project.openshift.io/v1/projects`
I0920 10:01:32.500360 1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/uscluster/apis/apps/v1/namespaces/openshift-console/deployments`
I0920 10:01:32.503337 1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/uscluster/apis/authorization.k8s.io/v1/selfsubjectaccessreviews`
2022/09/20 10:01:34 http: proxy error: context canceled
I0920 10:03:03.027486 1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/uscluster/apis/apps/v1/namespaces/openshift-console/deployments`
2022/09/20 10:03:03 http: proxy error: context canceled
I0920 10:03:03.529456 1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/uscluster/apis/project.openshift.io/v1/projects`
2022/09/20 10:03:03 http: proxy error: context canceled
```
Screen Shot 2022-09-20 at 18 02 52

**Comment by TheRealJon** [comment]
@yapei This is likely because your managed cluster has not been imported through ACM or the cluster name in ACM is different from the cluster name in the managed cluster config consumed by the console back end. So, in order for multicluster to work when running bridge locally, you'll have to follow these steps:

- Provision or use existing cluster with ACM installed
- Provision or use an existing second cluster as the managed cluster
- Follow the instructions in the `contrib/multicluster-environment.sh` script to init your local multicluster environment variables
- Import the managed cluster to the ACM cluster, using the same cluster name found in the BRIDGE_MANAGED_CLUSTERS env var
- Set `BRIDGE_K8S_MODE_OFF_CLUSTER_MANAGED_CLUSTER_PROXY` env var to the public cluster proxy route (on ACM cluster `oc get route cluster-proxy-addon-user -n multicluster-engine`)

**Comment by yapei** [comment]
tested again today
- Enable multi cluster on Hub cluster
- Import one managed cluster with name 'test-cluster-one'
- Visit 'test-cluster-one' from All Clusters dropdown, check console pod logs and we can see that console backend makes managed cluster API server requests through the cluster proxy that the MCE operator provides
```
I0921 13:34:38.657877       1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/test-cluster-one/apis/project.openshift.io/v1/projects/yapei-test-one-project`
I0921 13:34:39.231052       1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/test-cluster-one/api/v1/namespaces/yapei-test-one-project/resourcequotas`
I0921 13:34:39.569416       1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/test-cluster-one/apis/quota.openshift.io/v1/namespaces/yapei-test-one-project/appliedclusterresourcequotas`
I0921 13:34:39.573411       1 proxy.go:105] PROXY: `https://cluster-proxy-addon-user.multicluster-engine.svc:9092/test-cluster-one/apis/apps/v1/namespaces/yapei-test-one-project/deployments`
```
/label qe-approved

File: pkg/serverconfig/types.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ type Helm struct {
 	ChartRepo HelmChartRepo `yaml:"chartRepository"`
 }
 
-// ManagedClusterAPIServerConfig enables proxying managed cluster API server requests
+// TODO Remove this type once the console operator has been updated. It is obsolete now that we are using the MCE cluster proxy.
```
Comment (jhadvig) [inline pkg/serverconfig/types.go:154]: We should create a story for removing that logic for console operator

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
@TheRealJon looks like the PR needs rebase due to conflicts

**Review by jhadvig — APPROVED** [review]
/lgtm

**Pull request merged**